tests/manage/z_cluster/test_must_gather_options.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import pytest
+import tempfile
+import os
+from subprocess import TimeoutExpired
+
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import ManageTest, tier2
+from ocs_ci.ocs.must_gather.const_must_gather import GATHER_COMMANDS_VERSION
+from ocs_ci.utility.utils import mirror_image, exec_cmd
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.exceptions import CommandFailed
+
+logger = logging.getLogger(__name__)
+
+
+class TestMustGatherTimeoutOptions(ManageTest):
+
+    temp_folder = None
+
+    def setup(self):
+        assert "KUBECONFIG" in os.environ or os.path.exists(
+            os.path.expanduser("~/.kube/config")
+        ), "Cannot find $KUBECONFIG or ~/.kube/config; skipping log collection"
+        self.temp_folder = tempfile.mkdtemp()
+
+    def teardown(self):
+        logger.info(f"Delete must gather folder {self.temp_folder}")
+        if os.path.exists(path=self.temp_folder):
+            exec_cmd(cmd=f"rm -rf {self.temp_folder}")
+
+    @tier2
+    @pytest.mark.parametrize(
+        argnames=["should_pass", "cmd_param", "timeout"],
+        argvalues=[
+            pytest.param(*[True, "request-timeout", "1200s"]),
+            pytest.param(*[True, "request-timeout", "20m"]),
+            pytest.param(*[True, "request-timeout", "1h"]),
+            pytest.param(*[True, "timeout", "1200"]),
+            pytest.param(*[False, "timeout", "-1200"]),
+            pytest.param(*[False, "timeout", "-1200s"]),
+            pytest.param(*[False, "timeout", "-20m"]),
+            pytest.param(*[False, "timeout", "-1h"]),
+        ],
+    )
+    @pytest.mark.skipif(
+        float(config.ENV_DATA["ocs_version"]) not in GATHER_COMMANDS_VERSION,
+        reason=(
+            "Skipping must_gather test, because there is not data for this version"
+        ),
+    )
+    def test_must_gather_timeout(self, should_pass, cmd_param, timeout):
+        """
+        Test must-gather timeout parameter
+        """
+        pass_fail = "fail"
+        if should_pass:
+            pass_fail = "pass"
+        logger.info(
+            f"Running must-gather command with '{cmd_param}={timeout}'. Operation should {pass_fail}."
```

Comment:
If the main use of pass_fail variable is to print the right value in "Operation Should ..." then we can consider eliminating it by changing the should_pass to string to hold "Pass"/"Fail" instead of boolean and considering to rename it to 'expected_result'. Assert statement below can be changed accordingly too. This may help in removing if statement and pass_fail setting. Please check if this is possible to do.

Reply:
Done